src/pages/studyView/table/FixedHeaderTable.tsx

```diff
@@ -78,32 +99,53 @@ export default class FixedHeaderTable<T> extends React.Component<IFixedHeaderTab
         this.initDataStore();
     }
 
-    componentWillReceiveProps(nextProps: any) {
+    componentWillReceiveProps(nextProps:  IFixedHeaderTableProps<T>) {
         this.updateDataStore(nextProps);
     }
 
-    updateDataStore(nextProps: any) {
+    updateDataStore(nextProps: IFixedHeaderTableProps<T>) {
+        const tableDataStore = new FixedHeaderTableDataStore(
+            () => {
+                return this.props.data;
+            },
+            this.props.fixedTopRowsData || []
+        );
         this._store.setProps({
             columns: nextProps.columns,
-            data: nextProps.data,
+            dataStore: tableDataStore,
             initialSortColumn: this._sortBy,
             initialSortDirection: this._sortDirection
         });
     }
 
     initDataStore() {
+        const tableDataStore = new FixedHeaderTableDataStore(
+            () => {
+                return this.props.data;
+            },
+            this.props.fixedTopRowsData || []
+        );
         this._store = new LazyMobXTableStore<T>({
             columns: this.props.columns,
-            data: this.props.data,
+            dataStore: tableDataStore,
             initialSortColumn: this._sortBy,
             initialSortDirection: this._sortDirection
         });
     }
 
     @autobind
-    rowClassName({index}: any) {
+    rowClassName({ index }: any) {
         if (index > -1 && this.isSelectedRow(this._store.dataStore.sortedFilteredData[index])) {
-            return classnames(styles.row, styles.highlightedRow);
+            const classNames: string[] = [styles.row]
+            if (this.props.selectedRowClassName) {
+                const className = this.props.selectedRowClassName(this._store.dataStore.sortedFilteredData[index]);
+                if (className !== null) {
+                    classNames.push(className);
+                }
+            } else {
+                classNames.push(styles.highlightedRow);
```

Comment:
a bit confusing terms - using both highlighted and selected to describe the same thing, right? maybe the prop should be called `highlightedRowClassName`?